Src/Fido2.Models/Objects/AuthenticationExtensionsClientInputs.cs

```diff
@@ -20,7 +20,12 @@ public sealed class AuthenticationExtensionsClientInputs
     /// </summary>
     [JsonPropertyName("appid")]
     [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
-    public string AppID { get; set; }
+    public string AppID { private get; set; }
```

Comment:
Why did you put the getter private and add a method to retrieve the value of the app id?

Reply:
Same here @aseigler -- Not sure about this change.

Reply:
The attribute `[JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]` is there to avoid the prop to be serialized if unset. The private getter should not be necessary.

Reply:
According to the conformance test tool, AppID should not be part of the response at all. There is a specific testcase for this.

Reply:
In the latest conformance tool there is a testcase under this test group:
[Server-ServerPublicKeyCredentialGetOptionsResponse-Req-1 Test server generating ServerPublicKeyCredentialGetOptionsResponse]
**P-1** Get ServerPublicKeyCredentialGetOptionsResponse, and check that: (a) ....  

It fails on this if we let the appid property serialized:
AssertionError: Response.extensions MUST be set to the requested extensions! Expected "{"example.extension.bool":true,**"appid":"https://localhost:44329**"}" to equal "{"example.extension.bool":true}": expected { Object (example.extension.bool, **appid**) } to deeply equal { 'example.extension.bool': true }
    at n.eval (eval at compileCode (js/sandbox.js:25:26), <anonymous>:6750:16)
    
It is the first test of  "Attestations" / "GetAssertion Request".

![image](https://github.com/passwordless-lib/fido2-net-lib/assets/22753969/4ad59b8c-8447-410a-9595-2e982918b22f)

What is your suggestion?

Reply:
The lib's readme claims that it has 100% pass rate in conformance testing but it clearly fails in a lot of cases.
If that's important to you then we should figure out every case in a way that is acceptable for you.

Reply:
> The lib's readme claims that it has 100% pass rate in conformance testing but it clearly fails in a lot of cases. If that's important to you then we should figure out every case in a way that is acceptable for you.

That 100% pass was point in time.  We had done tons of work to get it to that point and were really excited.  Last I checked, the conformance tool had at least 2 or 3 server tests that are completely unpassable and have been for a few years.

Reply:
What do you mean by unpassable?
At certain testcases they have questionable requirements, but not unpassable.

My .net standard port passes all of them:
https://github.com/googyi/fido2-net-lib/tree/net-standard-conversion

Reply:
I just ran the latest version of the tool (v1.7.17) against your fork and I see 143 pass, 18 fail.

By unpassable, I mean like [Server-ServerAuthenticatorAttestationResponse-Resp-9].  The positive tests have been hopelessly broken for years.  Another example, we removed token binding a while back but it is still tested for in the tool.  I can't remember the last time I have seen the 100% button light up legitimately.

Reply:
![image](https://github.com/passwordless-lib/fido2-net-lib/assets/22753969/9ada1c07-1419-41d2-8882-1bb481fe816d)
I have done the same right now. Have you downloaded the needed conformance metadata and copied it to a conformance subfolder?
![image](https://github.com/passwordless-lib/fido2-net-lib/assets/22753969/a133ef93-526b-4f17-ac99-3249889fd2b5)

Reply:
Do you have a private branch with a bunch of tweaks that you are running this on?  I don't see how you are possibly legitimately passing these without heavy changes and special workarounds.  One of the fundamental issues with the way the tool works is the tests are a simple pass/fail -- which sounds simple on the surface, but for the failure case you actually have to be failing for the right reason, which you can only verify (in my experience) by menu -> open inspector and watching the responses to make sure they all line up with the appropriate test

<img width="575" alt="image" src="https://github.com/passwordless-lib/fido2-net-lib/assets/6605560/cea48d23-d85f-4c46-8560-7646fffc497a">

Reply:
I think everything is public in my fork and you should use the "net-standard-conversion" branch.
This is my latest commit in that branch:
![image](https://github.com/passwordless-lib/fido2-net-lib/assets/22753969/91c421c7-08f8-40a1-87c6-94bb5abf60e7)
Just check that you don't have this [Obsolete("Use property ResidentKey.")] stuff in the code.

I have ported the state of this .net core pull request and I readded your tokenbinding implementation.
And there was a lot work with asn stuff, serialization and cert verification, etc, but your old v2 version helped a lot in this.

"which you can only verify (in my experience) by menu -> open inspector"
I agree. So far I have not seen any testcase where it failed because of the wrong reason.
These tests have no big value, unless you want to get their certification. Then you have to do what they want.

Reply:
That branch definitely looks like it is passing everything legitimately.  Agree with your assessment in the value.  I wouldn't mind trying to get the implementation certified if some organization wanted to sponsor it.  I think we broke some of the FIDO tests implementing new WebAuthn features and moving away from the third party ASN.1 and CBOR libraries we were using.

Reply:
I have pushed a few changes to this branch to reach 100%.
I don't remember any ASN.1 or CBOR related thing that broke anything.

Reply:
@dotnet-policy-service agree company="LastPass"